Mod1.Lection2.Hw1/Mod2.Lection2.Hw1/Repositories/CartRepository.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Mod2.Lection2.Hw1.Models;
+
+namespace Mod2.Lection2.Hw1.Repositories;
+
+internal class CartRepository
+{
+    private readonly Product _product; // В конструкторе принимаем заполненный экземпляр Product
+
+    public CartRepository(Product product)
+    {
+        _product = product; // Сохраняем переданный экземпляр Product
+    }
+
+    public void AddtoCart(Cart cart)
+    {
+        while (true)
+        {
+            cart.SelectedProduct = Console.ReadLine(); // Здесь устанавливаем SelectedProduct через свойство
```

Comment:
Repositories are usually meant for just storing data and related logic. Not user interactions.